frontend/lib/providers/artefact_builds.dart

```diff
@@ -70,4 +70,25 @@ class ArtefactBuilds extends _$ArtefactBuilds {
 
     await _updateStateReviewDecision(testExecutionId, testExecution);
   }
+
+  Future<void> rerunTestExecution(int testExecutionId) async {
+    final api = ref.read(apiProvider);
+    await api.rerunTestExecution(testExecutionId);
+
+    final artefactBuilds = await future;
+    final newArtefactBuilds = <ArtefactBuild>[];
+    for (final ab in artefactBuilds) {
+      final newTestExecutions = [
+        for (final te in ab.testExecutions)
+          if (te.id == testExecutionId)
+            te.copyWith(isRerunRequested: true)
+          else
+            te,
+      ];
+
+      newArtefactBuilds.add(ab.copyWith(testExecutions: newTestExecutions));
+    }
+
+    state = AsyncData(newArtefactBuilds);
+  }
```

Comment:
I remember implementing this `_getUpdatedArtefactBuild` ([link](https://github.com/canonical/test_observer/blob/baefb5009ca6dcffa60eb27cca76b01949b81413/frontend/lib/providers/artefact_builds.dart#L18)) which seems to have the same behaviour as the code here. Perhaps we can either re-use it or refactor that one to match this one?